src/helpers/validations.js

```diff
@@ -0,0 +1,84 @@
+import _ from 'lodash'
+import { gql } from 'react-apollo'
+
+export const userValidations = (client) => ({
+  username: [
+    ({ username = '' }) =>
+      username.length >= 3 || 'must be at least 3 characters long',
+    ({ username = '' }) =>
+      /^[a-z0-9][a-z0-9-]*[a-z0-9]$/.test(username) ||
+      'must consist of a-z, A-Z, 0-9, -',
```

Comment:
`'must consist of a-z, 0-9, - and not start/end with a dash'`
(No upper case letters, not dash at beginning/end)